### ISSUE TITLE
Issues with assigning Modded mobs to config

### ISSUE THREAD

**Issue by Darkmega18** [SoraZodia/Cannibalism#24]
I've been looking into trying to get modded mobs (who don't extend a form of base minecraft mob naturally) to be added to cannibalism's jsons. However despite writing them in seemingly properly (using the name given by the dev knife, and adding the mod id properly etc) they do not work at all.

This isn't the case for vanilla mobs, as I've VERY easily been able to add skeletons, wolves, llamas and all sorts from vanilla game (and by extension their nonvanilla counter parts like say quark's hellhounds of special mob's special variants of zombies etc). But when I try to do this one for example for grimoire of gaia:

	{
	"entityID":"grimoireofgaia.satyress",
	"modID":"grimoireofgaia",
	"drops":"vanillafoodpantry:meat_portion#7",
	"minDamage":"4.0",
	"maxDamage":"5.0"
	}

I get no error, it just doesn't work properly for that particular mob... 
grimoireofgaia.satyress is what the knife tells me to put in, since it's not from vanilla. Then I put the modid in, which is also grimoireofgaia based on ingame and also from crash reports. and do the drop. but when I cut into that mob, there is no effect. or particles appear but no effect happens.

Do you know what could be going wrong with it?

**Comment by Darkmega18** [comment]
scratch that. polar bears from vanilla seem troublesome when I tried to add them in just the same way as others.

**Comment by SoraZodia** [comment]
I'll look into it. Just to be sure, are you on Minecraft 1.12.2? I don't recall that version having polar bears.

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by Darkmega18** [comment]
yeah, 1.12.2. I for some reason have polar bears and they're marked as minecraft related. I didn't think I was meant to have them either honestly, but seems I do, since vanilla food pantry gets meat from them. :V

**Comment by SoraZodia** [comment]
There was an issue with the knifes misinterpreting the mob's name/id. That same issue also messed up the dev knife's output, and its wording was not too clear in general. When it printed "grimoireofgaia.satyress", it meant that "grimoireofgaia" was the modID and "satyress" was the entityID; the correct json would had been:

```
{
"entityID":"satyress",
"modID":"grimoireofgaia",
"drops":"vanillafoodpantry:meat_portion#7",
"minDamage":"4.0",
"maxDamage":"5.0"
}
``` 

Those issues should be fixed now in 5.6.7. As for the dev knife's output, it will now always print out in the format of <modID>:<entityID>, even for vanilla mobs; as implied, <modID> would be for the modID field in the JSON and <entityID> would be for the entityID field.

Granted, polar bears are still seen as vanilla mobs, although that is not causing any trouble for the config now. I'm almost certain that they didn't exist in 1.12.2, so my best guess at this point is that they are from another mod, and were registered under the namespace used for vanilla mobs by said mod.

**Comment by Darkmega18** [comment]
alright, thanks for that. I'll be able to fiddle with more mobs who have drops and stuff now that I know. :D
